### ISSUE TITLE
Demo crash with RigidBody::Dynamic

### ISSUE THREAD

**Issue by fgadaleta** [jcornaz/heron#255]
I get this 
```
thread 'Compute Task Pool (0)' panicked at 'assertion failed: proxy.aabb.maxs[dim] >= self.min_bound', /home/frag/.cargo/registry/src/github.com-1ecc6299db9ec823/rapier3d-0.11.1/src/geometry/broad_phase_multi_sap/sap_axis.rs:55:13

```

**Comment by jcornaz** [comment]
I don't reproduce it yet.

Can you please provide more information? The exact commit at which you tried. What is your OS, rust version, and rendering backend?

Don't hesitate to provide any additional information you can think of.

**Comment by fgadaleta** [comment]
I made the ball collide with the ground and it crashes
Please find it here https://github.com/fgadaleta/herontest

**Comment by jcornaz** [comment]
I see [here](https://github.com/fgadaleta/herontest/blob/main/Cargo.toml#L27) that you are using the 3d version.

But that example is a 2d example that creates collision shapes without any depth: 
* https://github.com/fgadaleta/herontest/blob/main/src/main.rs#L37
* https://github.com/fgadaleta/herontest/blob/main/src/main.rs#L63

And rapier (not heron) crashes when the collision shape has a volume `<= 0`.

To solve your problem you can either:
* Use non-zero depth for the collision shapes
* Use the `2d` feature flag instead of the `3d`

**Comment by jcornaz** [comment]
I leave this issue open for discussion about how we could improve the situation.

On top of my head I think we could:
* ask rapier if it would be possible to not crash for empty volume
* fail faster, when the shape is created, rather than when it collide

**Comment by fgadaleta** [comment]
That also happens in 3D where there should be a volume already. For the record, this crashes too:


```
                    commands
                    .entity(entity)
                    .insert_bundle((transform, GlobalTransform::default()))
                    .insert(RigidBody::Dynamic)
                    .insert(CollisionShape::Sphere { radius: 2.7 })
                    .insert(PhysicMaterial {
                        restitution: 0.7,
                        density: 1.0,
                        ..Default::default()
                    })
                    .insert(CollisionLayers::new(Layer::Player, Layer::Player))
                    .insert(Velocity::from(Vec2::X * 300.0)
                            .with_angular(AxisAngle::new(Vec3::Y, -3.14)))
                    .insert(Drone);

```

**Comment by jcornaz** [comment]
> That also happens in 3D where there should be a volume already. For the record, this crashes too:

This is only the ball, you also need to give a depth to the ground.

**Comment by fgadaleta** [comment]
> I see [here](https://github.com/fgadaleta/herontest/blob/main/Cargo.toml#L27) that you are using the 3d version.
> 
> But that example is a 2d example that creates collision shapes without any depth:
> 
>     * https://github.com/fgadaleta/herontest/blob/main/src/main.rs#L37
> 
>     * https://github.com/fgadaleta/herontest/blob/main/src/main.rs#L63
> 
> 
> And rapier (not heron) crashes when the collision shape has a volume `<= 0`.
> 
> To solve your problem you can either:
> 
>     * Use non-zero depth for the collision shapes
> 
>     * Use the `2d` feature flag instead of the `3d`

you are right. Changing flag to `2d` does not crash and it actually works (the green ball crashes but in a good way lol)

**Comment by fgadaleta** [comment]
In my example I spawn a dynamic body like so:


```
                    commands
                    .entity(entity)
                    .insert_bundle((transform, GlobalTransform::default()))
                    .insert(RigidBody::Dynamic)
                    .insert(CollisionShape::Cuboid {
                        half_extends: Vec3::new(1., 1., 1.),
                        border_radius: Some(1.),
                    })
                    .insert(PhysicMaterial {
                        restitution: 0.7,
                        density: 1.0,
                        ..Default::default()
                    })
                    .insert(CollisionLayers::new(Layer::Player, Layer::Player))
                    .insert(Velocity::from(Vec2::X * 300.0)
                            .with_angular(AxisAngle::new(Vec3::Y, -3.14)))
                    .insert(Drone);
```


and all the rest is static, like so: 

```
                commands
                .entity(entity)
                .insert(RigidBody::Static)
                .insert(CollisionShape::Cuboid {
                    half_extends: Vec3::new(1., 1., 1.),
                    border_radius: Some(1.),
                })
                .insert(PhysicMaterial {
                    restitution: 30.,
                    friction: 1.0,
                    density: 100.0,
                    ..Default::default()
                })
                .insert(CollisionLayers::new(Layer::World, Layer::World))
                ;
```

**Comment by jcornaz** [comment]
> Collision is logged but ball disappears from screen (without actually colliding).

Due to the fast angular velocity, the body gets moved along the z-axis. I am not sure of the exact reason, but that is probably due to f32 imprecision in rapier's matrix multiplications.

Nevertheless, it does not crash. And if you remove the angular velocity or increase the depth of the ground it interacts correctly. Though it will always look strange, because these are 2d sprites allowed to move and rotate along the z-axis

And anyway, the `demo` is a 2d example, so I think it is pointless to try making it good in 3d. Though, I acknowledge, that we should clarify that the demo is not meant to run as-is in 3d.

**Comment by jcornaz** [comment]
> In my other reply I have been mentioning the 3d scenario, in which I have exactly the same problem.

Can you please provide more information? At least a full reproducer, and the full stack trace you get?

**Comment by fgadaleta** [comment]
Absolutely.
Check https://gitlab.com/fgadaleta/bevy_experiment (gltf included from LFS) 

If you just `cargo run` you will see it crashes.
If you change `src/world/level1.rs:1148` to `RigidBody::Static` it will run.

Let me know if there's something missing as this is a repo to reproduce the error only

**Comment by bit-garden** [comment]
We should have a note encouraging to check Rapiers common mistakes article, or a friendlier way to hint to what's wrong.  [this](https://rapier.rs/docs/user_guides/rust/common_mistakes/#the-simulation-panics) had the answer to start with. A 3d object with 0 in one dimension has 0 mass/volume.
